src/dataset/FFHQDataset.py

```diff
@@ -0,0 +1,253 @@
+from src.dataset.Dataset import Dataset
+from src.util.FileJar import FileJar
+
+import os, json
+import pandas as pd
+from tqdm import tqdm
+from multiprocessing import Pool, cpu_count
+import numpy as np
+
+
+class FFHQDataset(Dataset):
+    """
+    A subclass to Dataset which provides the ability to extract labels
+    from the dataset. These labels are saved to disk which allows subsequent
+    creations of the dataset to be performed without extraction the labels
+    again.
+
+    Before creating a FFQDataset, a directory should
+    exist with the name of "FFHQ_`resolution`" in the dataset directory.
+    There should also exist appriopriate directories for the labels ("label/")
+    and images ("image/").
+    """
+
+    # Dataset name
+    DS_NAME = "FFHQ"
+
+    # Subdirectory name to the images
+    DS_DIR_IMAGES = "image/"
+
+    # Subdirectory name to the labels
+    DS_DIR_LABELS = "label/"
+
+    # Filename of the extracted labels
+    DS_LABELS_NAME = "labels.pkl"
+
+    def __init__(self, resolution: int = 256):
+        """
+        Constructor which extracts labels from the dataset if not prevously done
+        in the super class with the help of `init_files()`,
```

Comment:
how could this be done previously? There exists no such method without an instance I believe.

Reply:
(also, "prevously" -> "previously")